fakeredis/_server.py

```diff
@@ -2401,6 +2404,24 @@ def on_connect(self, connection):
         pass
 
 
+# Redis <3.2 will not have a selector
+if hasattr(redis, 'selector'):
+    base_selector = redis.selector.BaseSelector
+else:
+    class FakeBaseSelector:
+        def __init__(self, sock):
+            pass
+    base_selector = FakeBaseSelector
+
+
+class FakeSelector(base_selector):
+    def check_can_read(self, timeout):
+        return True
```

Comment:
That doesn't look right. It should do something similar to `FakeSocket.can_read` (obviously sharing the common code).

Reply:
I'm not seeing a `FakeSocket.can_read`, there is a `FakeConnection.can_read` is that what you were referring to? That appears to handle ensuring the socket is connected, the selector handles checking if the socket has data: https://github.com/andymccurdy/redis-py/blob/3.2.0/redis/selector.py#L49 and https://github.com/andymccurdy/redis-py/blob/3.2.0/redis/selector.py#L96

Reply:
That's the one I meant.

Your `check_can_read` isn't checking if there is data available though, it's just returning true. It should check if `FakeSocket.responses.qsize()` is non-zero to determine whether there is data available, and wait up to the timeout for it to become non-zero.

I see that `FakeConnection.can_read` doesn't actually have a sleep in its `while True` loop; I'd intended to sleep for a short time (maybe 0.01 seconds).

Reply:
ping @bmerry, I pushed a fix for this.